client/src/transitions/SlideHorizontal.vue

```diff
@@ -1,38 +1,54 @@
 <!-- Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS -->
 <template>
-  <transition :name="!props.reverseDirection ? 'slide-left' : 'slide-right'">
+  <transition :name="getName()">
     <slot />
   </transition>
 </template>
 
 <script setup lang="ts">
+import { Position } from '@/transitions/types';
 import { defineProps } from 'vue';
 
 const props = defineProps<{
-  reverseDirection?: boolean;
+  from: Position;
```

Comment:
I'm not a big fan of having two props to define the transition behavior, because it delegates the configuration to the caller.

Which means you get a lot of logic in the caller:
```html
<slide-horizontal
  :from="state === HomePageState.OrganizationList ? Position.Left : Position.Right"
  :to="state === HomePageState.OrganizationList ? Position.Right : Position.Left"
>
```
and it can lead to weird things:
```html
<slide-horizontal
  :from="Position.Left"
  :to="Position.Left"
>
```
Why not define instead a `TransitionDirection` as `LeftToRight` and `RightToLeft`? It reduces the amount of logic, it's easier to understand, and it's less prone to errors. It does require a bit more logic in the component itself but that's ok.